### DISCUSSION TITLE
Slack通知先チャンネルに設定したいチャンネルが表示されない

### DISCUSSION THREAD

**Discussion by akhtkm** [topotal/waroom-roadmap#19] [Bug Report]
事象

1. 会社のSlackに通知用のチャンネルを作成した
2. OrganizationのSettingsで`通知チャンネルの設定`から `チャンネルを選択`プルダウンの一覧に 1. のチャンネルが表示されない
3. 右側にある更新ボタンを押すも効果なし
4. シークレットブラウザ等で試すも変化なし

推測
- 会社のSlackのチャンネル数が2700を超えているのでプルダウンリストの上限に達しているかもしれない？

提案
- 検索窓などから絞り込みができれば良いかもしれないです．

**Comment by sawa-zen** [comment]
@akhtkm 
ありがとうございます。現在 1000 件までを想定した作りとなっているため一覧に表示されなくなっていると思われます。
こちらで対応方法を検討いたしますのでいましばらくお待ちくださいませ 🙏🏻

**Reply by sawa-zen** [reply]
@akhtkm 
お待たせいたしました 🙏🏻 
実装側を修正するには時間がかかってしまいそうなので、もしよろしければ通知を飛ばしたいチャンネルの ID を指定していただければこちらで手動反映しようと思います。
GitHub discussions 上ではオープンすぎかと思うので `support@waroom.com` 宛にメールで送信いただけますと幸いです 🙇🏻

**Reply by sawa-zen** [reply]
@akhtkm 
チャンネルの ID は以下の手順で確認できます。

### 1. チャンネルの設定を押下
<img width="350" alt="スクリーンショット_2022-12-01_15_37_44" src="https://user-images.githubusercontent.com/3971271/204983197-127ce6e6-04fe-410e-9c57-df388b995039.png">

### 2. モーダルの一番下までスクロール
<img width="350" alt="スクリーンショット_2022-12-01_15_38_35" src="https://user-images.githubusercontent.com/3971271/204983423-c7628e10-b5b5-4cf6-a028-060361f07e82.png">

**Reply by sawa-zen** [reply]
メール受け取りました。ありがとうございます！対応いたしますので少々お待ち下さい。

**Reply by akhtkm** [reply]
よろしくお願いいたします！

**Reply by sawa-zen** [reply]
@akhtkm 
こちら対応が遅くなってしまっており申し訳ありません。
ただいま手動反映の準備を整えており来週の月曜中(12/5) に実行予定です。

**Reply by sawa-zen** [reply]
@akhtkm 
お待たせいたしました！手動反映完了いたしました 🙏🏻 
ご指定いただいたチャンネルに Waroom bot からメッセージが送信されたかと思いますが、届いていますでしょうか？

**Reply by akhtkm** [reply]
@sawa-zen 
届きました！ありがとうございます！
試しに作成もできていて，良さそうです．ご対応ありがとうございます 🙇🏼 
![image](https://user-images.githubusercontent.com/86997223/205565466-b61c512a-ec0c-4bb0-abc9-56e8fb0520e9.png)
![image](https://user-images.githubusercontent.com/86997223/205565481-9149d9d8-7a2e-49a9-a9e6-b47d955a901c.png)

**Reply by sawa-zen** [reply]
@akhtkm 
ご確認ありがとうございます！無事動作したようで安心しました 😌
1000件以上でも選択できるようになりましたらまたご連絡いたします。

**Reply by sawa-zen** [reply]
@akhtkm 
大変お待たせいたしました！チャンネルが1000件以上ある場合でも選択できるように修正いたしました 🙏🏻 
もしよろしければ組織の設定画面からチャンネル一覧が1000件以上表示できているか確認いただいてもよろしいでしょうか？

**Reply by akhtkm** [reply]
@sawa-zen 
ご対応ありがとうございます！
当初表示されてなかったチャネルに加え，他にも見えるようになったと思います 👏🏼 
（めちゃ縦スクロールになるので単純比較はできず 🙇🏼 ）

**Reply by sawa-zen** [reply]
@akhtkm 
確認ありがとうございます！大変助かりました 🙇🏻 今後名前検索ができるようにするなど選択UIの改善は進めていこうと思います 👍🏻

**Reply by akhtkm** [reply]
@sawa-zen はい，ヒアリングもOKです．少しだけ触っている状態なのでそこまでフィードバックできるかは不安ですが！
（ってあれ！消えちゃった？）

**Reply by sawa-zen** [reply]
@akhtkm 
ありがとうございます！ここでアポとってしまうのはよろしくないと思い修正してしまいました 🙇🏻 続きはメールにてご連絡します！
